projects/dotcms-field-elements/src/components/dot-binary-file/dot-binary-file.tsx

```diff
@@ -0,0 +1,312 @@
+                        type="text"
+                        onBlur={() => this.blurHandler()}
+                        disabled={this.disabled}
+                        placeholder={this.placeholder}
+                        onKeyDown={(event: KeyboardEvent) => this.keyDownHandler(event)}
+                        onKeyPress={(event: KeyboardEvent) => event.preventDefault()}
+                    />
+                    <input
+                        aria-describedby={getHintId(this.hint)}
+                        class={getErrorClass(this.status.dotValid)}
+                        disabled={this.disabled}
+                        id={getId(this.name)}
+                        onChange={(event: Event) => this.changeHandler(event)}
+                        required={this.required || null}
+                        type="file"
+                        accept={this.accept}
+                        value={this.value}
+                    />
+                    <button disabled={this.disabled} onClick={() => this.buttonHandler()}>
+                        {this.buttonLabel}
+                    </button>
+                </dot-label>
+                {getTagHint(this.hint)}
+                {this.errorElement}
+            </Fragment>
+        );
+    }
+
+    private dragAndDropHandler(): void {
+        this.el.addEventListener('dragover', (evt: DragEvent) => {
+            if (!this.disabled) {
+                this.el.classList.add('dot-dragover');
+            }
+            evt.preventDefault();
+        });
+
+        this.el.addEventListener('dragleave', (evt: DragEvent) => {
+            this.el.classList.remove('dot-dragover');
+            evt.preventDefault();
+        });
+
+        this.el.addEventListener('drop', (evt: DragEvent) => {
+            evt.preventDefault();
+            if (!this.disabled) {
+                this.el.classList.add('dot-dop');
+                this.el.classList.remove('dot-dragover');
+                const droppedFile: File = evt.dataTransfer.files[0];
+                if (this.isFileAllowed(droppedFile.name)) {
+                    this.setValue(droppedFile);
+                    this.setFileDisplayName(droppedFile.name);
+                } else {
+                    this.clearField(this.validationMessage);
+                }
+            }
+        });
+    }
+
+    // only supported in iOS.
+    private pasteHandler(): void {
+        this.el.addEventListener('paste', (evt: ClipboardEvent) => {
+            const clipboardData: DataTransfer = evt.clipboardData;
+            if (clipboardData.items.length) {
+                if (this.isPastingFile(clipboardData)) {
+                    this.handleFilePaste(clipboardData.items);
+                } else {
+                    this.handleURLPaste(clipboardData.items[0]);
+                }
+            }
+        });
+    }
+
+    private changeHandler(event: Event): void {
+        const input: HTMLInputElement = event.srcElement as HTMLInputElement;
```

Comment:
<img width="519" alt="Screen Shot 2019-06-21 at 3 45 44 PM" src="https://user-images.githubusercontent.com/751424/59953093-cd4f0500-943b-11e9-8c9a-f8d715a559b8.png">

Can you use target?